core/src/desul/atomics/Compare_Exchange_OpenMP.hpp

```diff
@@ -8,7 +8,18 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #ifndef DESUL_ATOMICS_COMPARE_EXCHANGE_OPENMP_HPP_
 #define DESUL_ATOMICS_COMPARE_EXCHANGE_OPENMP_HPP_
 #include "desul/atomics/Common.hpp"
-#include<cstdio>
+#include <cstdio>
+#include <omp.h>
+
+static constexpr bool omp_on_host() { return true; }
+
+#pragma omp begin declare variant match(device = {kind(host)})
+static constexpr bool omp_on_host() { return true; }
+#pragma omp end declare variant
```

Comment:
Is this really necessary since you already declared `omp_on_host` before?

Reply:
That's a good point. I am not sure if the host variant is needed when its implementation is similar to the base function. Maybe @crtrott knows more about the issue.